yt/yt/server/queue_agent/queue_static_table_exporter.cpp

```diff
@@ -241,45 +316,378 @@ class TQueueExportTask
     //! Original chunk specs fetched from master.
     std::vector<TChunkSpec> ChunkSpecs_;
     //! Pointers to chunk specs for chunks that are going to be exported within this iteration.
-    std::vector<const TChunkSpec*> ChunkSpecsToExport_;
-    //! Corresponds to the actual output table created.
-    TUserObject DestinationObject_;
-    TTransactionPtr UploadTransaction_;
-    //! Data statistics collected from attaching chunks.
-    TDataStatistics DataStatistics_;
+    TMap<ui64, std::vector<const TChunkSpec*>> ChunkSpecsToExportByUnixTs_;
 
     static constexpr TStringBuf ExportProgressAttributeName_ = "queue_static_export_progress";
     static constexpr TStringBuf ExportDestinationAttributeName_ = "queue_static_export_destination";
     static constexpr TStringBuf ExporterAttributeName_ = "queue_static_exporter";
 
-    ui64 GetMinFragmentUnixTs(TTimestamp timestamp)
+    struct TQueueExportSubtask final
     {
-        auto period = ExportConfig_.ExportPeriod.Seconds();
-        YT_VERIFY(period > 0);
+        const NNative::IClientPtr Client;
+        const NNative::IConnectionPtr Connection;
+
+        //! Corresponds to the queue being exported.
+        const TYPath Queue;
+        const TUserObject& QueueObject;
+        const TTableSchemaPtr& QueueSchema;
+        const TMasterTableSchemaId QueueSchemaId;
+
+        const TQueueStaticExportConfig ExportConfig;
```

Comment:
Well, I thought that some values are better of copied rather than stored as a pointer or a reference to the actual data, e.g. primitive types. To be completely honest, I'm not that sure how far this argument extends, and at what point do we prefer references or pointers rather than a copy.
In the case of `TQueueStaticExportConfig` I thought that maybe reference/pointer would be better, but at this point should this really be a lite yson struct? If we care enough to save some space for this config, shouldn't we make it ref-counted yson struct?

Reply:
Are you against making a copy of `TMasterTableSchemaId` (just a guid)? And did you meaningfully add this comment for the first line? Is there a problem with `const TTableSchemaPtr& QueueSchema;`?